client/jetpack-cloud/index.js

```diff
@@ -34,8 +34,9 @@ const redirectToPrimarySiteLanding = ( context ) => {
 	debug( 'controller: redirectToPrimarySiteLanding', context );
 	const state = context.store.getState();
 	const currentUser = getCurrentUser( state );
+	const isPrimarySiteJetpackSite = getPrimarySiteIsJetpack( state );
 
-	isJetpackSite( state, currentUser.primary_blog )
+	isPrimarySiteJetpackSite
```

Comment:
`isJetpackSite` relies on the list of all sites to verify if a given site is a Jetpack site. Very often, the list of sites is not in the app state yet, and so the `isJetpackSite` selector returns null, even when the site is in fact a Jetpack site.

For this reason, I'm replacing the `isJetpackSite` selector with `getPrimarySiteIsJetpack`. The latter uses a property that is part of the User object to verify if the user's primary site is a Jetpack site. This means that we no longer need the list of all user's sites to check if the primary site is a Jetpack site.